### PR TITLE
Add support for scipy sparse, fixing sparse tests

### PR DESCRIPTION
**Pull request by tomlaube**

### CODE REVIEW

**Comment by mrocklin** [comment]
Thanks for following up on this by the way.   I'm glad to see improved support for sparse data here.

File: dask_xgboost/tests/test_core.py [hunk 1]
```diff
@@ -98,56 +99,63 @@ def test_dmatrix_kwargs(c, s, a, b):
...
     dbst = yield dxgb.train(c, param, dX, dy)  # we can do this twice
 
-    dtrain = xgb.DMatrix(X, label=y)
-    bst = xgb.train(param, dtrain)
+    predictions = dxgb.predict(c, dbst, dX)
+    assert isinstance(predictions, da.Array)
 
-    result = bst.predict(dtrain)
-    dresult = dbst.predict(dtrain)
+    predictions_result = yield c.compute(predictions)._result()
```
[the remark this answers is not in the document]
Comment (tomlaube) [reply dask_xgboost/tests/test_core.py]: removed

File: dask_xgboost/core.py [hunk 2]
```diff
@@ -51,8 +53,10 @@ def concat(L):
         return np.concatenate(L, axis=0)
     elif isinstance(L[0], (pd.DataFrame, pd.Series)):
         return pd.concat(L, axis=0)
+    elif ss and isinstance(L[0], ss.spmatrix):
+        return ss.vstack(L, format='csr')
     elif sparse and isinstance(L[0], sparse.SparseArray):
-        return sparse.concatenate(L[0], axis=0)
+        return sparse.concatenate(L, axis=0)
```
[the remark this answers is not in the document]
Comment (tomlaube) [reply dask_xgboost/core.py]: So dask array uses this concatenate_lookup to dispatch the method that should be picked based on the containers it's build from?

[the remark this answers is not in the document]
Comment (mrocklin) [reply dask_xgboost/core.py]: Yes.  That function isn't used on dask arrays, it's used internally whenever we call `da.concatenate`.  Dask array faces the same problem that you face here, choosing which concatenate function to use based on the internal containers.  It's the same as the if/elif tree here, but it's extendable outside of the codebase as a dispatchable function.

**Comment by mrocklin** [comment]
OK, merging.  Thanks for following up @tomlaube !

**Comment by mrocklin** [comment]
> I'm curious, have you used this in practice? If so, how well or poorly does it perform?

I'm also still curious about this by the way.  If you try things out please report back :)

**Pull request merged**